### ISSUE TITLE
can't compare iterations

### ISSUE THREAD

**Issue by aquibbaig** [distributed-system-analysis/pbench-dashboard#180]
<img width="1438" alt="Demo" src="https://user-images.githubusercontent.com/26324376/125618238-69d76eee-9a9f-4a88-982d-7d2611b1d91c.png">
URL: http://pbench.perf.lab.eng.bos.redhat.com/dashboard/#/dashboard/comparison

**Comment by aquibbaig** [comment]
@dbutenhof @portante Can any of you help me debugging this elasticsearch query that we make when we call `fetchIterationSamples` query([link here](https://github.com/distributed-system-analysis/pbench-dashboard/blob/aed912abea99ffe87e07295fd46bfbfef8f5ede3/src/services/dashboard.js#L114)). Just double checking if it's correct because for every request it is results in an empty array of hits for me, which  is weird.

**Comment by dbutenhof** [comment]
It might simply be that there is no data. What's your endpoints.js?
Specifically, (I assume you're using the RDU2 ES7 server), what prefix are
you using?

The jam-pbench instance, for example, has no result-data-sample indices.
The staging-pbench instance however (I don't even know where this comes
from) does, and (in my Postman query) for run document
ID 15ed6b30e0f92df4fffdba7bae72c41d I see 20 hits on that query.

So I don't think it's the query that's the problem.

**Comment by aquibbaig** [comment]
my endpoints are at

```
'/dev/datastoreConfig': {
    "elasticsearch": "http://elasticsearch.perf.lab.eng.bos.redhat.com:9280",
    "results": "http://pbench.perf.lab.eng.bos.redhat.com",
    "graphql": "http://graphql.perf.lab.eng.bos.redhat.com:4466",
    "result_index": "v4.result-data.",
    "run_index": "v4.run.",
    "prefix": "dsa-pbench."
  },
```

I agree that there might not be any data, so I need to handle that case accordingly

**Comment by aquibbaig** [comment]
So, if the queries aren't a problem, I will go ahead and handle the case where there are no hits

**Comment by aquibbaig** [comment]
fixed by #195
